### PR TITLE
Waiter command Final PR

### PR DESCRIPTION
**Pull request by ARQAWA**

### CODE REVIEW

File: djangoheads/management/commands/waiter.py [hunk 1]
```diff
@@ -0,0 +1,276 @@
...
+        return "".join(random.choices("0123456789abcdef", k=length))
+
+    @staticmethod
+    def _find_celery_app_import_path() -> Optional[str]:
+        """Find celery app import path."""
+        work_dir = os.getcwd()
+        target_filename = "__init__.py"
+        celery_app_regex = re.compile(r"\s*celery_app\s*")
+
+        for root, _, files in os.walk(work_dir):
```
Comment (sergio-bershadsky) [inline djangoheads/management/commands/waiter.py:263]: Need to simplify

File: djangoheads/management/commands/waiter.py [hunk 2]
```diff
@@ -0,0 +1,276 @@
...
+            res = check_func()
+            if not res:
+                self._check_funcs.appendleft(check_func)
+                self._attempts -= 1
+                if self._attempts > 0:
+                    time.sleep(self._timeout)
+                    continue
+                sys.exit(1)
+
+    def check_db_read(self) -> bool:
```
Comment (sergio-bershadsky) [inline djangoheads/management/commands/waiter.py:80]: Every test should be presented as a separate file with same interface. That will help test better and define different usage entrypoints

File: djangoheads/management/commands/waiter.py [hunk 3]
```diff
@@ -0,0 +1,276 @@
...
+    def handle(self, *args: Tuple[Any, ...], **options: Dict[str, Any]) -> None:  # noqa: ARG002
+        """Command implementation."""
+        self._attempts = cast(int, options.get("n", self._attempts))
+        self._timeout = cast(int, options.get("t", self._timeout))
+
+        self._check_funcs.append(self.check_db_read)
+        # self._check_funcs.append(self.check_db_write) # TODO: Later we will add this check
+        self._check_funcs.append(self.check_migrations_are_applied)
+        self._check_funcs.append(self.check_caches)
+        self._check_funcs.append(self.check_celery)
```
Comment (sergio-bershadsky) [inline djangoheads/management/commands/waiter.py:67]: Need Standalone registry

File: pyproject.toml [hunk 4]
```diff
@@ -36,48 +37,17 @@ python = "^3.8"
 
 [tool.poetry.group.dev.dependencies]
 pre-commit = "^3.5.0"
-django = "^4.2.7"
+django = "^4.2.3"
```
Comment (sergio-bershadsky) [inline pyproject.toml:40]: Need understanding how to define deps for multi django libraries

**Review by sergio-bershadsky — CHANGES_REQUESTED** [review]

File: pyproject.toml (hunk 4 above)
Comment (ARQAWA) [reply pyproject.toml:40]: These deps using only in development environment, tests run inside nox environments on 4.1/4.2 djangos on python 3.8-3.12.

**Pull request closed without merging**